proof/invariant-abstract/CNodeInv_AI.thy

```diff
@@ -187,7 +187,7 @@ locale CNodeInv_AI =
         \<Longrightarrow> (ptr, nat_to_cref (zombie_cte_bits zbits) m) \<in> cte_refs (Zombie ptr zbits n) irqn"
   assumes finalise_cap_emptyable[wp]:
     "\<And>sl c f.
-      \<lbrace>emptyable sl and (invs and valid_mdb)\<rbrace>
+      \<lbrace>emptyable sl and invs\<rbrace>
```

Comment:
As a side benefit of collecting preconditions one at a time, it looks like `crunch` now generates simpler preconditions in some cases.

While collecting preconditions, `crunch` has a very simple test that tries to work out whether the preconditions being added are actually necessary. From memory this is mainly to minimise duplication in the preconditions.

In this case, the test would have previously involved trying to prove `emptyable sl s ⟶ invs s ∧ valid_mdb s` and failing. Now it ends up successfully proving `emptyable sl s ∧ invs s ⟶ valid_mdb s`, and so doesn't collect the `valid_mdb` condition.